src/utils.cpp

```diff
@@ -764,5 +771,227 @@ bool executePreAction(const nlohmann::json& i_parsedConfigJson,
 
     return true;
 }
+
+std::string getTruncatedInvPath(const std::string& i_invPath)
+{
+    // Truncate /xyz/openbmc_project/inventory if present
+    if (i_invPath.compare(0, 30, constants::pimPath) == 0)
+    {
+        return (std::string(i_invPath.cbegin() + 30, i_invPath.cend()));
+    }
+    return i_invPath;
+}
+
+bool isInventoryPathFromJSON(const types::Path i_path,
+                             const nlohmann::json i_jsonObj)
+{
+    if (!i_jsonObj.contains("frus"))
+    {
+        return false;
+    }
+
+    // Revisit: If JSON has complete path with prefix
+    // /xyz/openbmc_project/inventory
+
+    auto l_truncatedPath = utils::getTruncatedInvPath(i_path);
+
+    const nlohmann::json& l_listOfFrus =
+        i_jsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    for (const auto& l_itemFRUS : l_listOfFrus.items())
+    {
+        for (const auto& l_fru : l_itemFRUS.value())
+        {
+            if (l_fru.find("inventoryPath") != l_fru.end() &&
+                l_truncatedPath == l_fru.value("inventoryPath", ""))
+            {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool isDbusInvPath(const std::string& i_path)
```

Comment:
Always use mapper to detect if something is valid D-Bus Object path, or make the call directly. With the return value you will know if it is valid or not.